change/@ni-nimble-components-df093e6a-7cba-46ed-8d5d-07670af90719.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```

Comment:
Large refactor so please move to draft based on current policy https://github.com/ni/nimble/blob/main/CONTRIBUTING.md#code-review